analysis/src/CompletionBackEnd.ml

```diff
@@ -771,27 +815,40 @@ and completionsGetTypeEnv2 ~debug (completions : Completion.t list) ~full ~opens
 
 and getCompletionsForContextPath ~debug ~full ~opens ~rawOpens ~pos ~env ~exact
     ~scope ?(mode = Regular) contextPath =
+  let envCompletionIsMadeFrom = env in
   if debug then
     Printf.printf "ContextPath %s\n"
       (Completable.contextPathToString contextPath);
   let package = full.package in
   match contextPath with
   | CPString ->
     if Debug.verbose () then print_endline "[ctx_path]--> CPString";
-    [Completion.create "dummy" ~env ~kind:(Completion.Value Predef.type_string)]
+    [
+      Completion.create "dummy" ~env
+        ~kind:(Completion.Value (Ctype.newconstr Predef.path_string []));
+    ]
   | CPBool ->
     if Debug.verbose () then print_endline "[ctx_path]--> CPBool";
-    [Completion.create "dummy" ~env ~kind:(Completion.Value Predef.type_bool)]
+    [
+      Completion.create "dummy" ~env
+        ~kind:(Completion.Value (Ctype.newconstr Predef.path_bool []));
+    ]
   | CPInt ->
     if Debug.verbose () then print_endline "[ctx_path]--> CPInt";
-    [Completion.create "dummy" ~env ~kind:(Completion.Value Predef.type_int)]
+    [
+      Completion.create "dummy" ~env
+        ~kind:(Completion.Value (Ctype.newconstr Predef.path_int []));
+    ]
   | CPFloat ->
     if Debug.verbose () then print_endline "[ctx_path]--> CPFloat";
-    [Completion.create "dummy" ~env ~kind:(Completion.Value Predef.type_float)]
+    [
+      Completion.create "dummy" ~env
+        ~kind:(Completion.Value (Ctype.newconstr Predef.path_float []));
+    ]
```

Comment:
These changes are actually less correct than what's already in the repo (fixed that here when we set up the editor tooling in the compiler repo), so these can be reverted, as in "continue using `Predef.type_<primitive>`.